examples/chemistry/brusselator_bounded.jl

```diff
@@ -0,0 +1,272 @@
+  (U, V)::Form0{X} # State variables.
+  (U2V, One)::Form0{X} # Named intermediate variables.
+  (U̇, V̇)::Form0{X} # Tangent variables.
+  (α)::Constant{X}
+  (F)::Parameter{X}
+  # A named intermediate variable.
+  U2V == (U .* U) .* V
+  # Specify how to compute the tangent variables.
+  U̇ == 1 + U2V - (4.4 * U) + (α * Δ(U)) + F
+  V̇ == (3.4 * U) - U2V + (α * Δ(U))
+  # Associate tangent variables with a state variable.
+  ∂ₜ(U) == U̇
+  ∂ₜ(V) == V̇
+end
+# Visualize. You must have graphviz installed.
+to_graphviz(BrusselatorDynamics)
+
+# This is a "discrete" Decapode, with no morphisms.
+# TODO: Create an example with values that are not source terms.
+BrusselatorBoundaries = @decapode begin
+  L::Constant
+end
+
+BrusselatorMorphism = @relation () begin
+  rbl(C, Cl)
+end
+
+BrusselatorCollage = collate(
+  BrusselatorDynamics,
+  BrusselatorBoundaries,
+  BrusselatorMorphism,
+  Dict(
+    :C => :U,
+    :Cl => :L))
+to_graphviz(BrusselatorCollage)
+
+Brusselator = BrusselatorCollage
+
+infer_types!(Brusselator)
+resolve_overloads!(Brusselator)
+to_graphviz(Brusselator)
+
+# TODO: Create square domain of approximately 32x32 vertices.
+s = loadmesh(Rectangle_30x10())
+scaling_mat = Diagonal([1/maximum(x->x[1], s[:point]),
+                        1/maximum(x->x[2], s[:point]),
+                        1.0])
+s[:point] = map(x -> scaling_mat*x, s[:point])
+s[:edge_orientation] = false
+orient!(s)
+# Visualize the mesh.
+GLMakie.wireframe(s)
+sd = EmbeddedDeltaDualComplex2D{Bool,Float64,Point2D}(s)
+subdivide_duals!(sd, Circumcenter())
+
+# Select the vertices along the left "wall" of the domain.
+min_x = minimum(x -> x[1], s[:point])
+max_x = maximum(x -> x[1], s[:point])
+left_wall_idxs = findall(x -> x[1] == min_x, s[:point])
+right_wall_idxs = findall(x -> x[1] == max_x, s[:point])
+function generate(sd, my_symbol; hodge=GeometricHodge())
+  op = @match my_symbol begin
+    :rbl => (x,y) -> begin
+      x[left_wall_idxs] .= y
+      x
+    end
+    :rbr => (x,y) -> begin
+      x[right_wall_idxs] .= y
+      x
+    end
+    :.* => (x,y) -> x .* y
+    x => error("Unmatched operator $my_symbol")
+  end
+  return (args...) -> op(args...)
+end
+
+# Create initial data.
+@assert all(map(sd[:point]) do (x,y)
+  0.0 ≤ x ≤ 1.0 && 0.0 ≤ y ≤ 1.0
+end)
+
+U = map(sd[:point]) do (_,y)
+  22 * (y *(1-y))^(3/2)
+end
+
+V = map(sd[:point]) do (x,_)
+  27 * (x *(1-x))^(3/2)
+end
+
+L = fill(1.0, length(left_wall_idxs))
+
+F₁ = map(sd[:point]) do (x,y)
+ (x-0.3)^2 + (y-0.6)^2 ≤ (0.1)^2 ? 5.0 : 0.0
+end
+GLMakie.mesh(s, color=F₁, colormap=:jet)
+
+F₂ = zeros(nv(sd))
+
+One = ones(nv(sd))
+
+constants_and_parameters = (
+  fourfour = 4.4,
+  threefour = 3.4,
+  α = 0.001,
+  L = L,
+  F = t -> t ≥ 1.1 ? F₂ : F₁)
+
+# Generate the simulation.
+gensim(expand_operators(Brusselator))
+sim = eval(gensim(expand_operators(Brusselator)))
+fₘ = sim(sd, generate)
+
+# Create problem and run sim for t ∈ [0,tₑ).
+# Map symbols to data.
+u₀ = construct(PhysicsState,
```

Comment:
This should be updated to use the `ComponentArray` constructor as in the other examples.